src/main/java/de/tum/in/www1/artemis/service/metis/conversation/ChannelService.java

```diff
@@ -468,4 +472,24 @@ private static Channel createDefaultChannel(@NotNull String channelName) {
     private boolean allowDuplicateChannelName(Channel channel) {
         return channel.getExercise() != null || channel.getLecture() != null || channel.getExam() != null;
     }
+
+    /**
+     * Generates the channel name based on the associated lecture/exercise/exam title and a corresponding prefix.
+     * The resulting name only contains lower case letters, digits and hyphens and has a maximum length of 30 characters.
+     * Upper case letters are transformed to lower case and special characters are replaced with a hyphen, while avoiding
+     * consecutive hyphens, e.g. "Example(%)name" becomes "example-name".
+     *
+     * @param prefix prefix for the channel
+     * @param title  title of the lecture/exercise/exam to derive the channel name from
+     * @return the generated channel name
+     */
+    private static String generateChannelNameFromTitle(@NotNull String prefix, Optional<String> title) {
+        String channelName = prefix + title.orElse("");
+        // [^a-z0-9]+ matches all occurrences of single or consecutive characters that are no digits and letters
+        channelName = channelName.toLowerCase().replaceAll("[^a-z0-9]+", "-");
```

Comment:
if I remember that correctly, we also need to replace multiple `-` instances here with a single `-` since we wanted to avoid repeating them in autogenerated names, right?

Reply:
Yes, and this is covered by the current regex as well, since it replaces everything that is not a digit or a number. Hence something like `---` would also be replaced with `-`.